independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/ClassPathElement.java

```diff
@@ -74,9 +74,11 @@ default ResolvedDependency getResolvedDependency() {
     Set<String> getProvidedResources();
 
     /**
-     * Whether this class path element contains local resources.
+     * Whether the content of this tree comes from an archive or not.
+     * <p>
+     * This is useful for instance when you want to determine if the resources can be updated in dev mode.
      */
-    boolean providesLocalResources();
```

Comment:
My comments were about `PathTree` API only. This is a layer on top of it where `PathTree` is an impl detail and so here the methods could reflect the meaning that's relevant in this context.